### PR TITLE
New FromCBOR instance for PState

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -110,6 +110,8 @@ constraints:
   -- ones they reuse the one from 'some', but there isn't e.g. a proper version
   -- constraint from dependent-sum-template (which is the library we actually use).
   , dependent-sum > 0.6.2.0
+  , algebraic-graphs < 0.7
```
Comment (newhoggy) [inline cabal.project:115]: `algebraic-graphs` is a `plutus` dependency and `plutus` does not compile with the latest version on Hackage.

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs [hunk 2]
```diff
@@ -163,6 +164,12 @@ instance CC.Crypto crypto => FromSharedCBOR (PState crypto) where
     _retiring <- fromSharedPlusLensCBOR (toMemptyLens _1 id)
     pure PState {_pParams, _fPParams, _retiring}
 
+instance
+  ( CC.Crypto crypto
+  , FromSharedCBOR (PState crypto)
+  ) => FromCBOR (PState crypto) where
+  fromCBOR = evalStateT fromSharedPlusCBOR mempty
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs]: We can fit it into one line and use `fromNotSharedCBOR`
```suggestion
instance (CC.Crypto crypto, FromSharedCBOR (PState crypto)) => FromCBOR (PState crypto)
  fromCBOR = fromNotSharedCBOR
```

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs [hunk 3]
```diff
@@ -57,6 +57,7 @@ import qualified Data.UMap as UM
 import GHC.Generics (Generic)
 import Lens.Micro (_1, _2)
 import NoThunks.Class (NoThunks (..))
+import Control.Monad.State.Strict (evalStateT)
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs]:
```suggestion
```

**Review by lehins — DISMISSED** [review]
Minor comment, otherwise looks good

File: eras/byron/ledger/executable-spec/test/Test/Byron/Spec/Ledger/AbstractSize/Properties.hs [hunk 4]
```diff
@@ -153,5 +153,5 @@ testTxHasTypeReps =
     "Test HasTypeReps instances"
     [ testCase "AbstractSize - example - TxIn" exampleTypeRepsTxIn,
       testCase "AbstractSize - example - Tx" exampleTypeRepsTx,
-      testProperty "AbstractSize and HasTypeReps - Tx*" propTxAbstractSize
+      testPropertyNamed "AbstractSize and HasTypeReps - Tx*" "propTxAbstractSize" propTxAbstractSize
```
Comment (lehins) [inline eras/byron/ledger/executable-spec/test/Test/Byron/Spec/Ledger/AbstractSize/Properties.hs]: Looks like you are missing `OverloadedString` extension. However, I'd rather leave byron codebase alone, unless a change is really necessary

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs (hunk 2 above)
Comment (newhoggy) [reply eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs]: Fixed!

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs (hunk 3 above)
Comment (newhoggy) [reply eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs]: Fixed!

File: eras/byron/ledger/executable-spec/test/Test/Byron/Spec/Ledger/AbstractSize/Properties.hs (hunk 4 above)
Comment (newhoggy) [reply eras/byron/ledger/executable-spec/test/Test/Byron/Spec/Ledger/AbstractSize/Properties.hs]: Reverted.

**Review by lehins — APPROVED** [review]
Looks good.

We'll deal with `-Wno-deprecations` later. here is the ticket: #2910

**Comment by JaredCorduan** [comment]
@newhoggy do you know if you needed to update the index state in this PR? We tried to fix the nix shell in a [follow up PR](https://github.com/input-output-hk/cardano-ledger/pull/2914), but we are now getting failing golden tests and would like to figure out our options for fixing master...

**Pull request merged**